packages/test/test-end-to-end-tests/src/test/incrementalSummaryContext.spec.ts

```diff
@@ -0,0 +1,243 @@
+					break;
+				}
+				default:
+					throw new Error("Unknown operation");
+			}
+		}
+	}
+
+	public createOp(content: string) {
+		const op: IOp = {
+			type: "blobStorage",
+			value: content,
+		};
+		this.submitLocalMessage(op);
+	}
+
+	protected onDisconnect() {}
+	protected applyStashedOp(content: any): unknown {
+		throw new Error("Method not implemented.");
+	}
+}
+
+/**
+ * Validates w
+ */
+describeNoCompat(
+	"Incremental summary context fields are properly populated",
+	(getTestObjectProvider) => {
+		let provider: ITestObjectProvider;
+		const dataObjectFactory = new TestFluidObjectFactory([
+			["abc", TestSharedObject.getFactory()],
+		]);
+		const runtimeOptions: IContainerRuntimeOptions = {
+			summaryOptions: { summaryConfigOverrides: { state: "disabled" } },
+		};
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			dataObjectFactory,
+			[[dataObjectFactory.type, Promise.resolve(dataObjectFactory)]],
+			undefined,
+			[innerRequestHandler],
+			runtimeOptions,
+		);
+
+		const createContainer = async (): Promise<IContainer> => {
+			return provider.createContainer(runtimeFactory);
+		};
+
+		async function createSummarizer(container: IContainer, summaryVersion?: string) {
+			const createSummarizerResult = await createSummarizerFromFactory(
+				provider,
+				container,
+				dataObjectFactory,
+				summaryVersion,
+				getContainerRuntimeApi(pkgVersion, pkgVersion)
+					.ContainerRuntimeFactoryWithDefaultDataStore,
+			);
+			return createSummarizerResult.summarizer;
+		}
+
+		beforeEach(async () => {
+			provider = getTestObjectProvider({ syncSummarizer: true });
+		});
+
+		it("works", async () => {
+			const container = await createContainer();
+			const datastore = await requestFluidObject<ITestFluidObject>(container, "default");
+			const dds = await datastore.getSharedObject<TestSharedObject>("abc");
+			dds.createOp("test data 1");
+			dds.createOp("test data 2");
+			dds.createOp("test data 3");
+
+			const summarizer = await createSummarizer(container);
+			await provider.ensureSynchronized();
+			await summarizeNow(summarizer);
+
+			dds.createOp("test data 4");
+			await provider.ensureSynchronized();
+			const { summaryTree } = await summarizeNow(summarizer);
+			assert(summaryTree.tree[".channels"].type === SummaryType.Tree, "expecting a tree!");
+			const dataObjectTree = summaryTree.tree[".channels"].tree[datastore.runtime.id];
+			assert(dataObjectTree.type === SummaryType.Tree, "tree!");
+			const dataObjectChannelsTree = dataObjectTree.tree[".channels"];
+			assert(dataObjectChannelsTree.type === SummaryType.Tree, "data store channels tree!");
+			const ddsTree = dataObjectChannelsTree.tree[dds.id];
+			assert(ddsTree.type === SummaryType.Tree, "dds tree!");
+			assert(ddsTree.tree["0"].type === SummaryType.Handle);
```

Comment:
Is there a way to modify these blobs so that they summarize as tree instead of handle? I am guessing not for this particular DDS but that would be a good test case to have.